### PR TITLE
EVG-20295 Add option to unschedule downstream triggered versions

### PR DESCRIPTION
**Pull request by hadjri**
EVG-20295

### Description
Proposal #1 from the ticket isn't easily viable since I confirmed that cron does not ignore `activate: false`, whereas project triggers do ignore it, which is the opposite of our requested behavior. After some thinking I decided the simplest solution would be to introduce a boolean option on project triggers, where if it is set, will create deactivated versions rather than activated ones. Since false is the base case this shouldn't affect any existing workflows. Will create UI ticket on approval.

### Testing
Tested in staging, manually set the field on a trigger in the DB, triggered a version and confirmed it was [deactivated](https://spruce-staging.corp.mongodb.com/version/annie_copy2_8be2548818e8858aae06ae82e873ac1985004248_284a2ebc84b73f5ba2d3ad5c36aa78c7/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC). Updated existing unit tests confirming all versions/builds/tasks get deactivated when `UnscheduleTasks` is set.
### Documentation
Updated the triggers section describing this new option.

### CODE REVIEW

**Review by malikchaya2 — CHANGES_REQUESTED** [review]
I agree with this approach. While it is very specific to this use case, I think it's still the simplest way of accomplishing it in a way that is still generic enough.

File: model/project_ref.go [hunk 1]
```diff
@@ -255,8 +255,9 @@ type TriggerDefinition struct {
 	DateCutoff        *int   `bson:"date_cutoff,omitempty" json:"date_cutoff,omitempty"`
 
 	// definitions for tasks to run for this trigger
-	ConfigFile string `bson:"config_file,omitempty" json:"config_file,omitempty"`
-	Alias      string `bson:"alias,omitempty" json:"alias,omitempty"`
+	ConfigFile      string `bson:"config_file,omitempty" json:"config_file,omitempty"`
+	Alias           string `bson:"alias,omitempty" json:"alias,omitempty"`
+	UnscheduleTasks bool   `bson:"unschedule_tasks,omitempty" json:"unschedule_tasks,omitempty"`
```
[the remark this answers is not in the document]
Comment (hadjri) [reply model/project_ref.go:260]: Sure, changed

**Review by malikchaya2 — APPROVED** [review]
LGTM once tests pass

**Comment by hadjri** [comment]
Created: https://jira.mongodb.org/browse/EVG-20650

**Pull request merged**